File: Backend-POS/POS.Main/POS.Main.Business.Order/Models/Order/OrderDetailResponseModel.cs
using POS.Main.Business.Order.Models.OrderItem;

namespace POS.Main.Business.Order.Models.Order;

public class OrderDetailResponseModel
{
    public int OrderId { get; set; }
    public int TableId { get; set; }
    public string TableName { get; set; } = string.Empty;
    public string ZoneName { get; set; } = string.Empty;
    public string OrderNumber { get; set; } = string.Empty;
    public string Status { get; set; } = string.Empty;
    public int GuestCount { get; set; }
    public decimal SubTotal { get; set; }
    public string? Note { get; set; }
    public DateTime CreatedAt { get; set; }
    public List<OrderItemResponseModel> Items { get; set; } = new();
}

File: Frontend-POS/RBMS-POS-Client/src/app/core/api/models/order-detail-response-model.ts
/* tslint:disable */
/* eslint-disable */
import { OrderItemResponseModel } from '../models/order-item-response-model';
import { OrderLinkedTableModel } from '../models/order-linked-table-model';
export interface OrderDetailResponseModel {
  createdAt?: string;
  guestCount?: number;
  isLinked?: boolean;
  items?: Array<OrderItemResponseModel> | null;
  linkedTables?: Array<OrderLinkedTableModel> | null;
  note?: string | null;
  orderId?: number;
  orderNumber?: string | null;
  primaryTableName?: string | null;
  secondaryTableNames?: Array<string> | null;
  status?: string | null;
  subTotal?: number;
  tableId?: number;
  tableName?: string | null;
  totalGuestCount?: number;
  zoneName?: string | null;
}
